src/script/descriptor.cpp

```diff
@@ -1516,6 +1516,10 @@ std::vector<std::unique_ptr<PubkeyProvider>> ParsePubkeyInner(uint32_t key_exp_i
         error = "No key provided";
         return {};
     }
+    if (IsSpace(str.front()) || IsSpace(str.back())) {
+        error = strprintf("Pubkey '%s' is invalid due to whitespace", str);
+        return {};
+    }
```

Comment:
This could also be a WIP formatted private key right?
Should use the general term "key" here. E.g. 
```c++
strprintf(key '%s is invalid due to an extra whitespace at %s of the string', key_str, (IsSpace(str.front())?"beginning" : "end"));`
```

Reply:
Yes, it could be a WIF private key. I will change it to use "key".

Reply:
Done.